### ISSUE TITLE
Two issues with `$` in function names (e.g., `shiny::tags$b`)

### ISSUE THREAD

**Issue by bersbersbers** [r-lib/lintr#1144]
1. False-positive `extraction_operator_linter` with `shiny::tags$b`:

```r
# This is `extract.R`. Run
# lintr::lint("extract.R", linters = lintr::extraction_operator_linter())
shiny::tags$b("Hi")
```

```
/home/bers/extract.R:3:12: warning: Use `[[` instead of `$`  to extract an element.
shiny::tags$b("Hi")
           ^
```

2. I'd like  to suppress the `missing_argument_linter` on `shiny::tags$b`, but I effectively have to suppress it on all functions named `b`:

```r
# This is `argument.R`. Run
# lintr::lint("argument.R", linters = lintr::missing_argument_linter(except = "tags$b")) # lint is not suppressed
# lintr::lint("argument.R", linters = lintr::missing_argument_linter(except = "b")) # lint is suppressed
shiny::tags$b("Hi", )
```

```
/home/bers/argument.R:4:19: warning: Missing argument in function call.
shiny::tags$b("Hi", )
                  ^
```

**Comment by AshesITR** [comment]
That's not a false positive IMO:

```
> class(shiny::tags)
[1] "list"
```

`with(shiny::tags, b("Hi"))` would be an option if you want to avoid turning off the linter and using `[[`

**Comment by bersbersbers** [comment]
> That's not a false positive IMO:
> 
> ```
> > class(shiny::tags)
> [1] "list"
> ```

Oh, I did not know that. Thanks, you are absolutely right:
https://shiny.rstudio.com/articles/tag-glossary.html

This `extraction_operator_linter` is still prone to many unncessary (I'd say) positives in a reasonably clean codebase, considering that the above URL gives a total of 47 examples of using `tags$xyz`. (I guess that part of the reason why it's non-default.)

Regarding issue number 2 (the `missing_argument_linter`), it would still be nice if one could suppress `tags$b` or even `shiny::tags$b` (I rarely use `library`). Ideally, one could pass a regex to except `shiny::tags\$\w+` or something like that.

**Comment by AshesITR** [comment]
```
> shiny::tags$b("a")
<b>a</b>
> shiny::tags$b("a", )
<b>a</b>
```

**Comment by MichaelChirico** [comment]
I would just turn off the extraction linter for your use case. the linter wants you to write

```
shiny::tags[["b"]]("a")
```

if you don't like that (I personally wouldn't write that!) the linter is not for you :)

**Comment by bersbersbers** [comment]
> Why would you want to do `tags$b("a", )`?

`tags$b` is not an ideal example, but other Shiny tags are. Basically, writing a lot of Python code as well, I have become used to trailing commas in multi-line expressions:

> Black will add trailing commas to expressions that are split by comma where each element is on its own line. This includes function signatures.

https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#trailing-commas

This allows one to easily add or remove elements from list. Compare the example from https://github.com/r-lib/lintr/issues/563#issue-741748895: if you have something like

```r
x <- list(
  a = 1,
  b = 2,
  c = 3,
)
```

(assuming this would be valid R code) then you can easily remove any parameter `a`, `b` or `c`, or add `d = 4,` anywhere, and you will always end of with valid R code. In addition, it's source-control friendly, since removals and additions do not change neighboring lines.

A lot of libraries support this, explicitly or implicitly, for many of their functions, such as `ggplot2` (compare https://github.com/tidyverse/ggplot2/issues/4237), `rmarkdown`, `glue`, `devtools`, etc. There's also code around to patch functions such as `c`, `list` etc., see https://www.r-bloggers.com/2013/03/r-and-the-last-comma/

**Comment by MichaelChirico** [comment]
I think we could add a parameter to missing_argument_linter allow_trailing to handle this case

**Comment by bersbersbers** [comment]
This would be beautiful :) (not urgent, though)

**Comment by MichaelChirico** [comment]
it will definitely come after release, unless you're keen to cook up a PR :)

**Comment by AshesITR** [comment]
Closing here for better discoverability of the feature request.
